### ISSUE TITLE
Kindle CS 12 will not display covers

### ISSUE THREAD

**Issue by Treishpel** [ciromattia/kcc#765]
Running KCC 6.2.1
As per the wiki I tried converting from a cbr/cbz with the kindle plugged in and connected. I used the Kindle CS 12 profile and converted to MOBI/AZW3.
![image](https://github.com/user-attachments/assets/3aaf0e8c-f19b-45f0-83a1-860af1ab61d2)

Tried direct upload, file still shows up on device with the default cover.
Also tried uploading the same file and also a different file that was done during the same conversion though Calibre. Calibre does show the cover, but on the device it's still the same default one.
Restarted device between "upload runs".

Sorry if it's a stupid post but.....am I missing something?

Thank you.

**Comment by axu2** [comment]
@Treishpel the new 12th Gen. Kindles need to use the kindle scribe cover method. It's linked in the faq in the readme.

**Comment by Treishpel** [comment]
@axu2 Oh, I saw scribe and older and thought maybe it was an older thing. I'll give it a shot and see. Thanks.

**Comment by Treishpel** [comment]
So I tried following [this](https://github.com/ciromattia/kcc/issues/508), I did the fake flash drive, I have the thumbnails in the system/thumbnails folder. But not too sure about the "Copy the thumbnails onto your Scribe in the appropriate location" I do not have a system folder in the kindle file structure. I'm on Windows.

**Comment by axu2** [comment]
Huh. Maybe the folder is hidden? Try opening this url in file explorer. (change the scribe part to whatever yours is) @Treishpel 

```
This PC\Kindle Scribe\Internal Storage\system\thumbnails
```

Otherwise, maybe amazon changed something on 12th gen?

**Comment by Treishpel** [comment]
![image](https://github.com/user-attachments/assets/e829e856-151a-4d5a-8c93-c0229a21f4ed)
This is the folder structure on it.

I have Hidden files and folders enabled and also "Hide protected operating system files" disabled
`This PC\Kindle Colorsoft Signature Edition\Internal Storage\system\thumbnails` does not go anywhere.

So looks like you might be right.

**Comment by axu2** [comment]
If Amazon changed that it'd be a tragedy. 

Follow this thread for more details. 

https://www.mobileread.com/forums/showthread.php?p=4464071

**Comment by axu2** [comment]
It's possible changing the metadata from EBOK to PDOC will make newer kindles read the cover directly from the file instead of from the thumbnails folder. 

- https://github.com/ciromattia/kcc/issues/685

**Comment by axu2** [comment]
@Treishpel does making a system/thumbnails folder work?

**Comment by Treishpel** [comment]
@axu2 nope. Context menu will not even display "new" option. Trying to drag and drop or copy paste the folder also did nothing. That's one of the first things i did i tried copying the documents and system folder from the "fake kindle" drive, it was popping up that documents already exists but nothing for system.

I did not try the PDOC vs EBOK yet.

**Comment by Treishpel** [comment]
Okay, So far the solution i found is to convert them. Then I edit metadata in bulk with calibre. Save to disk from calibre. And upload to device using https://www.amazon.com/sendtokindle.

Sometimes I need to compress the file in calibre if it's like a 2in1 volume and it exceeds 200MB as that is max upload limit.

**Comment by axu2** [comment]
@Treishpel I modified kcc to use pdoc instead of ebok, try using that instead and report back. Prebuild exe linked in 

- #767

**Comment by Treishpel** [comment]
I'll be honest, I'm pretty new to the whole ebook/ereader field, so I have no idea what the difference between MOBI and EPUB is. So far for the workaround I mentioned earlier, I've been doing EPUBs.

Now to get to your question, @axu2 I took two cbz volumes and converted to EPUB+MOBI with the version you linked, I drag-and-dropped one as EPUB and one as MOBI, I can confirm that the MOBI shows up on the device and it also has the cover. 
The EPUB however does not show up in my library which I find strange since with the "Send to kindle" site, I was doing EPUB, and EPUB is listed as a compatible filetype on the amazon product page, but maybe they are doing a conversion on their end...

TL;DR, Yes, converting to mobi with this version did work and cover is now displayed.

**Comment by axu2** [comment]
Yes, kcc works by creating an epub.

Then you either convert the epub to a mobi with kcc.

Or upload the epub via Send to Kindle where it's converted by amazon.

**Comment by Treishpel** [comment]
Well I'm very glad it's working now. I started converting a whole batch of 25 volumes. This makes it so much easier since there's no more waiting around to upload to amazon. 

The only extra step now is to edit metadata, which, completely off-topic, would love a way to bulk edit metadata inside KCC, I only do author now since I edit filenames before converting, but being able to let's say, choose a batch of CBZs and before conversion to be able to set author, publisher or whatever else might be needed.
This would allow to just get CBZ, convert to MOBI directly and then just drag and drop on device.

Anyways I strayed too far from the main topic, I guess with that fix this could be marked as Closed.

**Comment by axu2** [comment]
Open a separate issue for that, but indeed that sounds like a great feature for at least author.

**Comment by Treishpel** [comment]
I did not even hear about ComicTagger. I have converted more stuff with that version and so far everything seems to work fine. Right now after conversion I upload into Calibre change author but I have to resave the file. Maybe I'll give ComicTagger a try. Anyway. I'll open a new issue for the author since it would be a nice QOL feature.

**Comment by axu2** [comment]
@Treishpel does changing author in calibre cause issues like losing full screen?

**Comment by Treishpel** [comment]
Not sure what fullscreen would look like LOL. I was putting all the details in the other issue.
My current workflow is
-Convert with KCC
-Upload in calibre.
-Change metadata around
-Save to disk from calibre.
-Upload to device via usb

This is the end result.
![20241110_184940.jpg](https://github.com/user-attachments/assets/e70d6e81-cc9c-462c-ac29-c4dc123ed6c3)

And these are the options i seem to have available
![20241110_185054.jpg](https://github.com/user-attachments/assets/b114ae4a-eff4-486a-a363-d43a02ef6e64)

I have not seen any instability issues so far.
